File: activityloghelper/src/main/java/com/example/administrator/activityloghelper/model/MainModelFactory.java
package com.example.administrator.activityloghelper.model;

import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;
import android.content.Context;
import android.support.annotation.NonNull;

public class MainModelFactory implements ViewModelProvider.Factory {

    private final Context mContext;

    private MainModelFactory(Context context) {
        mContext = context;
    }

    public static ViewModelProvider.Factory get(Context context) {
        return new MainModelFactory(context);
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        if (modelClass.isAssignableFrom(AccessibilityViewModel.class)) {
            return (T) new AccessibilityViewModel(mContext);
        }
        throw new IllegalArgumentException("Unknown ViewModel class");
    }
}

File: activityloghelper/src/main/java/com/example/administrator/activityloghelper/model/MainModelFactory.kt
package com.example.administrator.activityloghelper.model;

import android.content.Context
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

class MainModelFactory(private val context: Context) : ViewModelProvider.Factory {

    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(AccessibilityViewModel::class.java)) {
            return AccessibilityViewModel(context) as T
        }
        throw IllegalArgumentException("Unknown ViewModel class")
    }

    companion object {
        @JvmStatic
        fun get(context: Context): ViewModelProvider.Factory {
            return MainModelFactory(context)
        }
    }

}
